stores/sql/mysql/migrations/main/migration_00030_host_sectors.sql

```diff
@@ -12,7 +12,26 @@ CREATE TABLE `host_sectors` (
   CONSTRAINT `fk_host_sectors_db_host` FOREIGN KEY (`db_host_id`) REFERENCES `hosts` (`id`) ON DELETE CASCADE
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
 
-INSERT IGNORE INTO host_sectors (updated_at, db_sector_id, db_host_id)
-SELECT NOW(), cs.db_sector_id, c.host_id
-FROM contract_sectors cs
-INNER JOIN contracts c ON cs.db_contract_id = c.id AND c.host_id IS NOT NULL;
+-- MySQL uses row-level locking with INNODB, so we use a loop to insert in
+-- batches to avoid running out of space in the locking table.
+SET @start_id = 0;
+SET @batch_size = 100000;
+SET @done = FALSE;
+
+WHILE NOT @done DO
+  INSERT INTO host_sectors (updated_at, db_sector_id, db_host_id)
+  SELECT NOW(), cs.db_sector_id, c.host_id
+  FROM contract_sectors cs
+  INNER JOIN contracts c ON cs.db_contract_id = c.id AND c.host_id IS NOT NULL;
+  WHERE cs.id > @start_id
```

Comment:
contract_sectors don't have an `id` field, but even if they would this loop wouldn't account for id holes or a table where the id for whatever reason starts at a number greater than zero